command/lock.go

```diff
@@ -66,22 +75,40 @@ Options:
   -name=""                   Optional name to associate with lock session.
   -token=""                  ACL token to use. Defaults to that of agent.
   -pass-stdin                Pass stdin to child process.
+  -try=timeout               Attempt to acquire the lock up to the given
+                             timeout (eg. "15s").
+  -monitor-retry=n           Retry up to n times if Consul returns a 500 error
+                             while monitoring the lock. This allows riding out brief
+                             periods of unavailability without causing leader
+                             elections, but increases the amount of time required
+                             to detect a lost lock in some cases. Defaults to 3,
+                             with a 1s wait between retries. Set to 0 to disable.
   -verbose                   Enables verbose output
 `
 	return strings.TrimSpace(helpText)
 }
 
 func (c *LockCommand) Run(args []string) int {
+	var lu *LockUnlock
+	return c.run(args, &lu)
+}
+
+// run exposes the underlying lock for testing.
+func (c *LockCommand) run(args []string, lu **LockUnlock) int {
```

Comment:
In this case `lu` gets created down in the function so it's a little messy to get it returned. I looked at adding `lu` as a return value but I would have to touch a ton of `return` statements, so this seemed like a good compromise.